proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,368 @@
+        assert isinstance(m.fs.properties._phase_component_set, Set)
+
+    @pytest.mark.unit
+    def test_parameters(self, frame_stateblock):
+        m = frame_stateblock
+        # test that the parameter variables are fixed
+        for v in m.fs.properties.component_objects(Var):
+            if not v.is_fixed():
+                raise Exception(
+                    "Variable {v_name} is unfixed, but all variables on the "
+                    "property parameter block must be fixed by default".format(
+                        v_name=v.name))
+
+    @pytest.mark.unit
+    def test_state_variables(self, frame_stateblock):
+        m = frame_stateblock
+        # state variables
+        state_vars_dict = m.fs.stream[0].define_state_vars()
+        # check minimum number of state variables (i.e. flow, temperature, pressure)
+        if len(state_vars_dict) < 3:
+            raise Exception(
+                "Property package has {num_state_vars} state variables, which is less "
+                "than the minimum of 3 (i.e. flow, temperature, and pressure)".format(
+                    num_state_vars=len(state_vars_dict)))
+
+        # check that state variables are in metadata
+        metadata = m.fs.properties.get_metadata().properties
+        for sv_name in state_vars_dict:
+            if sv_name not in metadata:
+                raise Exception(
+                    "State variable {sv_name} is not included in the "
+                    "metadata".format(sv_name=sv_name))
+
+    @pytest.mark.unit
+    def test_permanent_properties(self, frame_stateblock):
+        m = frame_stateblock
+        metadata = m.fs.properties.get_metadata().properties
+        for v_name in metadata:
+            if metadata[v_name]['method'] is None:
+                if not hasattr(m.fs.stream[0], v_name):
+                    raise Exception(
+                        "Property {v_name} is included in the metadata but is not found "
+                        "on the stateblock".format(v_name=v_name))
+
+    @pytest.mark.unit
+    def test_on_demand_properties(self, frame_stateblock):
+        m = frame_stateblock
+        metadata = m.fs.properties.get_metadata().properties
+
+        # check that properties are not built if not demanded
+        for v_name in metadata:
+            if metadata[v_name]['method'] is not None:
+                if m.fs.stream[0].is_property_constructed(v_name):
+                    raise Exception(
+                        "Property {v_name} is an on-demand property, but was found "
+                        "on the stateblock without being demanded".format(v_name=v_name))
+
+        # check that properties are built if demanded
+        for v_name in metadata:
+            if metadata[v_name]['method'] is not None:
+                if not hasattr(m.fs.stream[0], v_name):
+                    raise Exception(
+                        "Property {v_name} is included in the metadata but is not found "
```

Comment:
It might be worth rewording this error message to say that the property was not built when called for (i.e. be more specific so that this failure isn't confused with a similar one).